framework/snippets/src/network_response.rs

```diff
@@ -58,16 +59,72 @@ fn process_out(tx: &TransactionOnNetwork) -> Vec<Vec<u8>> {
     }
 }
 
+fn process_logs(tx: &TransactionOnNetwork) -> Vec<Log> {
+    if let Some(api_logs) = &tx.logs {
+        return api_logs
+            .events
+            .iter()
+            .map(|event| Log {
+                address: Address::from_slice(&event.address.to_bytes()),
+                endpoint: event.identifier.clone(),
+                topics: extract_topics(event),
+                data: extract_data(event),
+            })
+            .collect::<Vec<Log>>();
+    }
+
+    Vec::new()
+}
+
+fn extract_data(event: &Events) -> Vec<Vec<u8>> {
+    let mut out: Vec<Vec<u8>> = Vec::new();
+    if let Some(data) = event.data.clone() {
+        match data {
+            Data::String(string) => out.push(string.into_bytes()),
+            Data::Vec(vec) => return vec.into_iter().map(|s| s.into_bytes()).collect(),
+        }
+    }
+
+    out
+}
+
+fn extract_topics(event: &Events) -> Vec<Vec<u8>> {
+    event
+        .topics
+        .clone()
+        .unwrap_or_default()
+        .into_iter()
+        .map(|s| s.into_bytes())
+        .collect()
+}
+
 fn process_out_from_log(tx: &TransactionOnNetwork) -> Option<Vec<Vec<u8>>> {
     if let Some(logs) = &tx.logs {
         logs.events.iter().rev().find_map(|event| {
             if event.identifier == "writeLog" {
                 if let Some(data) = &event.data {
-                    let decoded_data = String::from_utf8(base64_decode(data)).unwrap();
-
-                    if decoded_data.starts_with('@') {
-                        let out = decode_scr_data_or_panic(decoded_data.as_str());
-                        return Some(out);
+                    match data {
+                        Data::String(string) => {
+                            let decoded_data = String::from_utf8(base64_decode(string)).unwrap();
```

Comment:
Duplicate code. Easier to just convert to vec first:
```
let data_vec = match data {
    Data::String(string) =>vec![string],
    Data::Vec(vec) => vec,
};

for data_member in vec.iter() { ...
```

Better still, create a method in Data that outputs a vec